manifests/init.pp

```diff
@@ -3,45 +3,48 @@
-  $cron_allow            = 'absent',
-  $cron_deny             = 'present',
-  $cron_allow_path       = '/etc/cron.allow',
-  $cron_allow_owner      = 'root',
-  $cron_allow_group      = 'root',
-  $cron_allow_mode       = '0644',
-  $cron_deny_path        = '/etc/cron.deny',
-  $cron_deny_owner       = 'root',
-  $cron_deny_group       = 'root',
-  $cron_deny_mode        = '0644',
-  $cron_d_path           = '/etc/cron.d',
-  $cron_hourly_path      = '/etc/cron.hourly',
-  $cron_daily_path       = '/etc/cron.daily',
-  $cron_weekly_path      = '/etc/cron.weekly',
-  $cron_monthly_path     = '/etc/cron.monthly',
-  $cron_dir_owner        = 'root',
-  $cron_dir_group        = 'root',
-  $cron_dir_mode         = '0755',
-  $cron_files            = undef,
-  $cron_allow_users      = undef,
-  $cron_deny_users       = undef,
-  $crontab_vars          = undef,
-  $crontab_tasks         = undef,
-  $periodic_jobs_content = undef,
-  $periodic_jobs_manage  = true,
-  $service_enable        = true,
-  $service_ensure        = 'running',
-  $service_name          = 'USE_DEFAULTS',
+  $package_ensure                = 'installed',
+  $package_name                  = 'USE_DEFAULTS',
+  $crontab_path                  = '/etc/crontab',
+  $crontab_owner                 = 'root',
+  $crontab_group                 = 'root',
+  $crontab_mode                  = '0644',
+  $cron_allow                    = 'absent',
+  $cron_deny                     = 'present',
+  $cron_allow_path               = '/etc/cron.allow',
+  $cron_allow_owner              = 'root',
+  $cron_allow_group              = 'root',
+  $cron_allow_mode               = '0644',
+  $cron_deny_path                = '/etc/cron.deny',
+  $cron_deny_owner               = 'root',
+  $cron_deny_group               = 'root',
+  $cron_deny_mode                = '0644',
+  $cron_d_path                   = '/etc/cron.d',
+  $cron_hourly_path              = '/etc/cron.hourly',
+  $cron_daily_path               = '/etc/cron.daily',
+  $cron_weekly_path              = '/etc/cron.weekly',
+  $cron_monthly_path             = '/etc/cron.monthly',
+  $cron_dir_owner                = 'root',
+  $cron_dir_group                = 'root',
+  $cron_dir_mode                 = '0755',
+  $cron_files                    = undef,
+  $cron_allow_users              = undef,
+  $cron_deny_users               = undef,
+  $crontab_vars                  = undef,
+  $crontab_tasks                 = undef,
+  $periodic_jobs_content         = undef,
+  $periodic_jobs_manage          = true,
+  $service_enable                = true,
+  $service_ensure                = 'running',
+  $service_name                  = 'USE_DEFAULTS',
   # deprecated
-  $enable_cron           = undef,
-  $ensure_state          = undef,
+  $enable_cron                   = undef,
+  $ensure_state                  = undef,
 ) {
 
+  # Hash crontabs from various hiera hierarchies
+  $user_crontabs = hiera_hash('cron::user_crontabs', undef)
```

Comment:
Sometimes users like to be able to decide that only the most specifc data should be pulled from hiera. This does allows them to have global values accross all hiera levels as default and stiff override them for a specific host.

See https://github.com/ghoneycutt/puppet-module-yum/blob/v1.4.0/manifests/init.pp#L101-L108 for an example that is very close to your work.

Reply:
Good point, I will put that in as an option.

Reply:
This has been fixed.